### PR TITLE
Introduce version constrain sieve pipeline unit

### PR DESCRIPTION
**Pull request by fridex**

### CODE REVIEW

**Comment by fridex** [comment]
recheck

**Comment by fridex** [comment]
> Just one question, should it be constraint? Otherwise LGTM!

Agh, right. I'll fix it :+1: thanks for spotting it!

**Comment by pacospace** [comment]
recheck

File: tests/sieves/test_version_constraint.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+from thoth.adviser.sieves import VersionConstraintSieve
+from thoth.adviser.exceptions import SieveError
+from thoth.python import Source
+from thoth.python import PackageVersion
+
+from ..base import AdviserTestCase
+
+
+class TestVersionConstrainSieve(AdviserTestCase):
```
Comment (pacospace) [inline tests/sieves/test_version_constraint.py]: Constraint

File: tests/sieves/test_version_constraint.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    """Test version constraint sieve."""
+
+    def test_configuration_error(self) -> None:
+        """Test removing a locked package based on direct dependencies."""
+        # Default values should error, no adjustment to values.
+        with pytest.raises(SieveError):
+            VersionConstrainSieve().pre_run()
```
Comment (pacospace) [inline tests/sieves/test_version_constraint.py]: Constraint, also below

**Comment by fridex** [comment]
PTAL @pacospace

**Pull request merged**